app/prompts/admin.py

```diff
@@ -0,0 +1,333 @@
+
+
+class PromptDetailsWidget(forms.Widget):
+    def render(self, name, value, attrs=None, renderer=None):
+        # self.instance is added by PromptForm
+        if not self.instance.id:
+            return ""
+
+        url = reverse(
+            get_prompt_viewname(self.instance),
+            kwargs={"object_id": self.instance.id},
+        )
+        return mark_safe(f'<a href="{url}">Edit details</a>')
+
+
+class PromptForm(IsDisabledCheckboxForm):
+    prompt_details = forms.CharField(
+        widget=PromptDetailsWidget(),
+        label="prompt details",
+        required=False,
+    )
+    prompt_type = forms.ChoiceField(
+        choices=[
+            ("", "Please select..."),
+            *PROMPT_LABELS.items(),
+        ]
+    )
+
+    class Meta:
+        model = Prompt
+        fields = "__all__"
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        if self.instance:
+            # Pass instance to widget to display link to specific Prompt
+            self.fields["prompt_details"].widget.instance = self.instance
+
+            # For existing prompts set prompt type
+            if self.instance.pk:
+                self.fields["prompt_type"].initial = type(
+                    self.instance
+                ).__name__
+                self.fields["prompt_type"].disabled = True
+                self.fields["prompt_type"].required = False
+
+
+class PromptFormSet(BaseInlineFormSet):
+    def clean(self):
+        super().clean()
+
+        # Check for enabled prompts exceeding limit
+        count = 0
+        for form in self.forms:
+            instance: TextPrompt = form.instance
+
+            if not instance.disabled_at:
+                if instance.text:
+                    count += 1
+
+                if count > ENABLED_PROMPT_LIMIT and instance.order is not None:
+                    form.add_error(
+                        "text",
+                        ValidationError(
+                            f"You cannot have more than {ENABLED_PROMPT_LIMIT} enabled prompts"
+                        ),
+                    )
+
+        disallow_duplicates(
+            self.forms, "order", "This order number is used in another prompt"
+        )
+
+
+class PromptAdmin(admin.TabularInline):
+    model = Prompt
+    form = PromptForm
+    formset = PromptFormSet
+    extra = 1
+    ordering = ["order"]
+    fields = [
+        "text",
+        "prompt_type",
+        "prompt_details",
+        "order",
+        "is_disabled",
+        "disabled_by",
+        "created_by",
+    ]
+    readonly_fields = [
+        "disabled_by",
+        "created_by",
+    ]
+    list_display = ["text", "feedback_form", "order"]
+    list_filter = ["feedback_form"]
+    search_fields = ["text", "uuid"]
+
+    def get_queryset(self, request):
+        query_set = super().get_queryset(request)
+        return query_set.select_subclasses(
+            BinaryPrompt, RangedPrompt, TextPrompt
```

Comment:
Is there any value having those model classes spelled out explicitly in `select_subclasses`? It does not feel like this to me. Thinking more that if there was ever a need to add a new prompt type in the future, we'd have more places to update like this.

Reply:
Nope, I already removed those in my latest work.